src/transformStateWithClones.js

```diff
@@ -6,8 +6,48 @@
  *
  * @return {Object[]}
  */
+function copyStateAndPush(Obj, arr) {
+  const copyObj = { ...Obj };
+
+  arr.push(copyObj);
+}
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const historyArray = [];
+  const copyState = { ...state };
+
+  for (const act in actions) {
+    const typeAction = Object.values(actions[act]);
+    const commandData = typeAction[1];
+
+    if (typeAction[0] === 'addProperties') {
+      for (const data in commandData) {
+        copyState[data] = commandData[data];
+      }
```

Comment:
You could use object spread here [Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#:~:text=prototype.map().-,Spread%20in%20object%20literals,-Copying%20and%20merging)